### ISSUE TITLE
Duplicate click events on Checkbox with `data-track` attribute in Matomo Tag Manager

### ISSUE THREAD

**Issue by cheton** [cheton/styled-ui#782]
An issue was reported that when using the `data-track` attribute on a `Checkbox` component results in the Matomo Tag Manager's track event listener receiving two click events, one from the text itself and one from the checkbox element. The main reason for this is that the Checkbox component is wrapped in a `label` tag, and when the text is clicked, it simultaneously changes the checkbox state, triggering two click events.

For example, the following approach binds the data-track attribute to the `<label>`:

```jsx
<Checkbox data-track="checkbox">
  Check Me
</Checkbox>
```

In our Matomo Tag Manager, the track event implementation is set to default to searching for the first matching `data-track` attribute in the parent hierarchy. However, since both click events are triggered within the `label`, Matomo records two checkbox click events.

![image](https://github.com/trendmicro-frontend/tonic-ui/assets/447801/57beb5b1-f8d9-4fe2-8bfa-10975901305d)

![image](https://github.com/trendmicro-frontend/tonic-ui/assets/447801/859df493-68ab-4c0d-9d55-75e524b182cb)

**Comment by cheton** [comment]
The ideal way is to include the `data-track` attribute in the input element, but currently `Checkbox` does not provide such a prop. Apart from modifying `Checkbox` to support `inputProps`, the workaround for now is to use `ref` to achieve it. An example is shown below:

```jsx
<Checkbox
  ref={(node) => {
    node.setAttribute('data-track', 'checkbox');
  }}
  checked={checked}
  onChange={handleChange}
>
  My checkbox
</Checkbox>
```
